networkit/cpp/generators/RmatGenerator.cpp

```diff
@@ -34,36 +175,11 @@ Graph RmatGenerator::generate() {
     count numEdges = n * edgeFactor * n * 1.0 / static_cast<double>(n - reduceNodes);
     count wantedEdges = (n - reduceNodes) * edgeFactor;
     Graph G(n - reduceNodes, weighted);
-    double ab = a + b;
-    double abc = ab + c;
-
-    auto quadrant([&]() {
-        double r = Aux::Random::probability();
-        TRACE("r: ", r);
-
-        if (r <= a) {
-            return 0;
-        } else if (r <= ab) {
-            return 1;
-        } else if (r <= abc) {
-            return 2;
-        } else
-            return 3;
-    });
-
-    auto drawEdge([&]() {
-        node u = 0;
-        node v = 0;
-        for (index i = 0; i < scale; ++i) {
-            count q = quadrant();
-            u = u << 1;
-            v = v << 1;
-            u = u | (q >> 1);
-            v = v | (q & 1);
-        }
+    // Reset the internal state of the alias table:
+    curBits = {0, 0};
+    remainingBits = 0;
 
-        return std::make_pair(u, v);
-    });
+    auto drawEdge([&]() { return sampleEdge(Aux::Random::getURNG(), (uint8_t)scale); });
```

Comment:
As above (casting). Also use `sampleEdge`, since `drawEdge` simply now calls this.